modules/enterprise/gui/coregui/src/main/java/org/rhq/coregui/client/inventory/resource/detail/monitoring/table/MetricsTableView.java

```diff
@@ -361,6 +365,18 @@ public void expandOpenedRows() {
         protected Canvas getExpansionComponent(final ListGridRecord record) {
             final Integer definitionId = record.getAttributeAsInt(METRIC_DEF_ID.getValue());
             final Integer resourceId = record.getAttributeAsInt(RESOURCE_ID.getValue());
+
+            MeasurementUnits recordUnits = null;
+            if (record.getAttributeAsString(DISPLAY_UNITS_NAME.getValue())!=null && record.getAttributeAsString
+                    (DISPLAY_UNITS_FAMILY.getValue())!=null) {
+
+                recordUnits = MeasurementUnits.getUsingDisplayUnits(
+                        record.getAttributeAsString(DISPLAY_UNITS_NAME.getValue()),
+                        MeasurementUnits.Family.valueOf(record.getAttributeAsString(DISPLAY_UNITS_FAMILY.getValue())));
+            }
+
+            final MeasurementUnits recordDisplayUnits = recordUnits;
```

Comment:
I can't see really from this github UI, but what's the point of this assignment?

Reply:
This is because I need to make if final to access it from the inner AsyncCallback anonymous object.

https://github.com/rhq-project/rhq/pull/328/files/c1a6281b8a0f33132df33a4dff6d71fc4bd79436#diff-07cbfda325e06a09752feca7adc9bbebR417